### PR TITLE
[WiP][Testing]Add sub menu icon parameter

### PR DESCRIPTION
**Pull request by VVeremjova**
PR previously was created and reviewed on #921.  
Please see @AByzhynar, @dcherniev, @AGritsevich, @Kozoriz, @dtrunov, @pvvasilev

Related to [ APPLINK-23379](https://adc.luxoft.com/jira/browse/APPLINK-23379)

### CODE REVIEW

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 1]
```diff
@@ -97,11 +121,9 @@ TEST_F(AddSubMenuRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
   Event event(hmi_apis::FunctionID::UI_AddSubMenu);
   event.set_smart_object(*ev_msg);
 
-  MockHmiInterfaces hmi_interfaces;
-  ON_CALL(app_mngr_, hmi_interfaces()).WillByDefault(ReturnRef(hmi_interfaces));
-  EXPECT_CALL(hmi_interfaces,
-              GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI))
-      .WillOnce(Return(am::HmiInterfaces::STATE_AVAILABLE));
+  //  EXPECT_CALL(hmi_interfaces_,
```
Comment (AGritsevich) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova, please delete it.

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 2]
```diff
@@ -58,18 +58,42 @@ using am::MockMessageHelper;
...
+    (*command_msg_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
+    (*command_msg_)[am::strings::params][am::strings::connection_key] =
+        kConnectionKey;
+
+    ON_CALL(app_mngr_, hmi_interfaces())
+        .WillByDefault(ReturnRef(hmi_interfaces_));
+
+    EXPECT_CALL(hmi_interfaces_,
+                GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI))
+        .WillRepeatedly(Return(am::HmiInterfaces::STATE_AVAILABLE));
```
Comment (pvvasilev) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova If you don't specify the times this call happens, why don't you change it to `ON_CALL`?

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 3]
```diff
@@ -132,6 +150,251 @@ TEST_F(AddSubMenuRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
...
+  EXPECT_CALL(app_mngr_settings_, app_storage_folder())
+      .WillOnce(ReturnRef(file_storage));
+
+  EXPECT_CALL(mock_message_helper_,
+              HMIToMobileResult(hmi_apis::Common_Result::WARNINGS))
+      .WillOnce(Return(mobile_apis::Result::WARNINGS));
+
+  MessageSharedPtr result_msg(
+      CatchMobileCommandResult(CallOnEvent(*command, event)));
+  const mobile_apis::Result::eType kReceivedResult =
```
Comment (pvvasilev) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @VVeremjova Local constants should be named as regular variables. Please check all tests.

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @pvvasilev See  820964e

**Comment by LuxoftAKutsan** [comment]
@VVeremjova  please check base. Probably it should not be develop

**Pull request merged**